src/index.ts

```diff
@@ -130,35 +130,40 @@ function wrapExportDefaultDeclaration(path: NodePath<any>) {
 
 function superJsonWithNext(): PluginObj {
   return {
-    name: 'replace gSSP',
+    name: 'add superjson to pages with prop getters',
     visitor: {
       Program(path, state) {
         const filename =
           getFileName(state) ?? nodePath.join('pages', 'Default.js');
         if (!filename.includes('pages' + nodePath.sep)) {
           return;
         }
+        const filesToSkip = ([] as string[]).concat(
+          ...['_app', '_document', '_error'].map((name) => [
+            name + '.js',
+            name + '.jsx',
+            name + '.ts',
+            name + '.tsx',
+          ])
+        );
+        if (filesToSkip.some((name) => filename.includes(name))) {
+          return;
+        }
 
         const body = path.get('body');
 
-        let foundGSSP = false;
```

Comment:
These lines were neccessary to make sure that we don't wrap every single page in the `withSuperJSON` HOCs. That's why the test case was there. What's the reasoning behind removing that?

Reply:
Did you PR description? 😅 (Also see my comments in the issue yesterday). This `foundGSSP` thing is broken because in the client build the prop getter code doesn't exist. I tried using a global variable to track the files that have prop getters, but this did not work (client and server builds have different scopes apparently).

So the only solution I could think of was add HOC to every page. Open to other ideas though!

Reply:
Okay, I guess I failed to connect the PR description to the actual change here. There was so much stuff in that one commit, failed to separate the important from the unimportant 😅

But I get it. If adding a HOC to every component solves it for now, so be it. We should look for other solutions though (but not in this PR)